### PR TITLE
run pylint recursively in github actions

### PR DESCRIPTION
**Pull request by ryanlong1004**
Description

The original command pylint *py does not search recursively.

Updated per https://stackoverflow.com/questions/36873096/run-pylint-for-all-python-files-in-a-directory-and-all-subdirectories.

Fixes https://github.com/ufs-community/workflow-tools/issues/22

Type of change
Update pylint command to act recursively.

How Has This Been Tested?
Running the pylint command via the CLI.

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.

Need to figure out what best practices are for linting a project.
The [documentation](https://pylint.pycqa.org/en/latest/user_guide/run.html) shows multiple ways.  The approach here is applying pylint recursively file-by-file.

**Comment by aerorahul** [comment]
Next obvious question is, should we be linting `docs/*.py` or restrict to `src/`

**Comment by ryanlong1004** [comment]
Once everything is moved in ```src```, calling ```pylint src``` will lint all python files  in the src package.

We're recursively looking for python files now because packages won't be introduced until the UW-44 branch is merged into develop.

**Review by christinaholtNOAA — DISMISSED** [review]
I think we should be linting all our python code, but that's just my preference. I think it's outside the scope of this PR to clean up the docs python code. Maybe we can open an issue for that one?

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# https://editorconfig.org/
```
Comment (christinaholtNOAA) [inline .editorconfig:1]: I think it's fine that this is added, but it does seem outside the scope of this PR, and will only help a subset of our code contributors: those using IDEs or those who have downloaded a plugin.

**Review by christinaholtNOAA — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (ryanlong1004) [reply .editorconfig:1]: @christinaholtNOAA sorry about that.  I was adding to my fork without realizing it would come up in the PR.

**Comment by ryanlong1004** [comment]
@christinaholtNOAA agreed.  

Starting a few [discussions](https://github.com/ufs-community/workflow-tools/discussions) to start talking about a project charter and best practices.

**Comment by christinaholtNOAA** [comment]
Ooh! I haven't used that tool yet. Very nice.
